web-app/packages/lib/src/common/permission_utils.ts

```diff
@@ -209,3 +217,22 @@ export function getProjectPermissionByRoleName(
   }
   return mapper[roleName]
 }
+
+export function calculateProjectPermission(
+  project_role: ProjectRoleName,
+  workspace_role: WorkspaceRoleName
+): ProjectRoleName {
+  const mappedWorkspaceRole: ProjectRoleName =
+    workspace_role === 'admin' ? 'owner' : (workspace_role as ProjectRoleName)
+
+  if (project_role === 'none' && workspace_role === 'guest') {
+    return 'none'
+  }
+
+  const projectRoleIndex = ROLE_HIERARCHY.indexOf(project_role)
+  const workspaceRoleIndex = ROLE_HIERARCHY.indexOf(mappedWorkspaceRole)
+
+  return projectRoleIndex > workspaceRoleIndex
```

Comment:
We need calculate project role only if it's null. If not null -> stay same as before. I think we do not need such logic.

Reply:
A workspace role higher than a project role is a valid scenario, no? See user id 6...

![Screenshot from 2024-12-18 11-56-41](https://github.com/user-attachments/assets/80da617d-d9d9-430f-bab0-cfe147043e43)

Reply:
We have deal with @varmar05 , that BE will return new role attribute which will be calculated role. We can then use it in dropdown and project_role will be used just for add / update project collaborator handling.